packages/baseui-codemods/src/v9-theme-scale.test.js

```diff
@@ -0,0 +1,256 @@
+*/
+
+// @flow
+
+import codemod from './v9-theme-scale.js';
+import {Fixture} from './test-utilities.js';
+
+async function transform(content) {
+  const fixture = new Fixture();
+  await fixture.write(content);
+  await codemod({dir: fixture.dir});
+  const transformed = await fixture.read();
+  await fixture.remove();
+  return transformed;
+}
+
+describe('shift theme usage for colors and fonts', () => {
+  describe('does update', () => {
+    it('font values in Blocks', async () => {
+      const content = `const Foo = () => (
+    <React.Fragment>
+      <Block font="font1100" />
+      <Block font="font1000" />
+      <Block font="font900" />
+      <Block font="font800" />
+      <Block font="font700" />
+      <Block font="font600" />
+      <Block font="font500" />
+      <Block font="font470" />
+      <Block font="font460" />
+      <Block font="font450" />
+      <Block font="font400" />
+      <Block font="font350" />
+      <Block font="font300" />
+      <Block font="font250" />
+      <Block font="font200" />
+      <Block font="font100" />
+    </React.Fragment>
+  );`;
+
+      expect(await transform(content)).toMatchInlineSnapshot(`
+        "const Foo = () => (
+            <React.Fragment>
+              <Block font=\\"font1450\\" />
+              <Block font=\\"font1050\\" />
+              <Block font=\\"font950\\" />
+              <Block font=\\"font850\\" />
+              <Block font=\\"font750\\" />
+              <Block font=\\"font650\\" />
+              <Block font=\\"font550\\" />
+              <Block font=\\"font450\\" />
+              <Block font=\\"font400\\" />
+              <Block font=\\"font350\\" />
+              <Block font=\\"font300\\" />
+              <Block font=\\"font250\\" />
+              <Block font=\\"font200\\" />
+              <Block font=\\"font150\\" />
+              <Block font=\\"font100\\" />
+              <Block font=\\"font100\\" />
+            </React.Fragment>
+          );"
+      `);
+    });
+
+    it('direct usage of font values', async () => {
+      const content = `const Foo = styled('div', props => ({
+    ...props.$theme.typography.font300
+  }));`;
+
+      expect(await transform(content)).toMatchInlineSnapshot(`
+        "const Foo = styled('div', props => ({
+            ...props.$theme.typography.font200
+          }));"
+      `);
+    });
+
+    it('direct usage of values with destructured `$theme`', async () => {
+      const content = `const Foo = styled('div', ({$theme}) => ({
+    ...$theme.typography.font300
+  }));`;
+
+      expect(await transform(content)).toMatchInlineSnapshot(`
+        "const Foo = styled('div', ({$theme}) => ({
+            ...$theme.typography.font200
+          }));"
+      `);
+    });
+
+    it('direct usage of values with destructured value', async () => {
+      const content = `const Foo = styled('div', ({$theme: {typography: {font300}}}) => ({
+    ...$theme.typography.font300
```

Comment:
is this testing what you intend? looks like it's referencing `$theme` which should throw if it was in source code, right?

```suggestion
    ...font300
```

Reply:
A similar, but equally important test would be `...typography.font300`

Reply:
ah yeah updated this and added the `typography` case